src/reconcile/rec_write.c

```diff
@@ -1843,11 +1854,13 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
             return (__wt_set_return(session, EBUSY));
 
         /* If we need to restore the page to memory, copy the disk image. */
-        if (r->cache_write_restore)
+        if (restore) {
```

Comment:
Isn't this equivalent to:
```
r->leave_dirty || F_ISSET(S2C(session), WT_CONN_IN_MEMORY) || r->page->type == WT_PAGE_COL_FIX;
```
In that case, you could remove the `restore` state on the `WT_MULTI`. I guess there's probably something missing but it seems like this pattern of propagating this state down the stack, aggregating it for a page and then passing it back up shouldn't be necessary (and it's confusing).

Reply:
No, it's not. leave dirty is at the original page's level. Here we want something to indicate whether the splitted pages are clean or not.

Reply:
Some splitted pages may be clean and some may be dirty. Ideally, we should be able to evict the clean pages and restore the dirty pages. This approach is to achieve that.

Reply:
Let's move the discussion to the new pr WT-5527

Reply:
Yeah, I understand what you're saying. `r->leave_dirty` is going to be true for every split page even if only some of the split pages end up dirty.